### PR TITLE
feat(core): load reorder from kmx+, test case 🙀

### PR DESCRIPTION
**Pull request by srl295**
- load reorder groups and their usets from KMX+
- add a test case matching the nod-Lana spec entry
(the interesting cases commented out for now)

For: #7375

@keymanapp-test-bot skip

~[I somehow picked up #9259 so will need to rebase once that merges]~ fixed it…

### CODE REVIEW

**Review by jahorton — APPROVED** [review]

**Review by jahorton — COMMENTED** [review]
Though, uh... those failed user tests are a bit of a concern.

File: core/src/ldml/ldml_transforms.cpp [hunk 1]
```diff
@@ -514,8 +537,21 @@ transforms::load(
...
+      reorder_group newGroup;
+
+      // fetch each reorder in the group
+      for (KMX_DWORD itemNumber = 0; itemNumber < group->count; itemNumber++) {
+        const kmx::COMP_KMXPLUS_TRAN_REORDER *reorder = tranHelper.getReorder(group->index + itemNumber);
+
+        element_list elements;
+        element_list before;
+
+        bool load_ok = elements.load(kplus, reorder->elements) && before.load(kplus, reorder->before);
```
Comment (jahorton) [inline core/src/ldml/ldml_transforms.cpp:549]: From one of the failed build-checks:

```
07:46:48 
  [26/34] Compiling C++ object src/libkmnkbp0.so.0.0.0.p/ldml_ldml_transforms.cpp.o
07:46:48 
  FAILED: src/libkmnkbp0.so.0.0.0.p/ldml_ldml_transforms.cpp.o
07:46:48 
  c++ -Isrc/libkmnkbp0.so.0.0.0.p -Isrc -I../../../src -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wextra -Werror -std=c++14 -O3 -fPIC -isystem../../../../common/include -isystem../../../include -isysteminclude -DKMN_KBP -DKMN_KBP_EXPORTING -Wctor-dtor-privacy -Wdouble-promotion -Wendif-labels -Wno-unknown-pragmas -Wno-missing-field-initializers -Wnon-virtual-dtor -Wshadow -fvisibility=hidden -fvisibility-inlines-hidden -MD -MQ src/libkmnkbp0.so.0.0.0.p/ldml_ldml_transforms.cpp.o -MF src/libkmnkbp0.so.0.0.0.p/ldml_ldml_transforms.cpp.o.d -o src/libkmnkbp0.so.0.0.0.p/ldml_ldml_transforms.cpp.o -c ../../../src/ldml/ldml_transforms.cpp
07:46:48 
  ../../../src/ldml/ldml_transforms.cpp: In static member function ‘static km::kbp::ldml::transforms* km::kbp::ldml::transforms::load(const km::kbp::kmx::kmx_plus&, const km::kbp::kmx::COMP_KMXPLUS_TRAN*, const km::kbp::kmx::COMP_KMXPLUS_TRAN_Helper&)’:
07:46:48 
  ../../../src/ldml/ldml_transforms.cpp:549:14: error: unused variable ‘load_ok’ [-Werror=unused-variable]
07:46:48 
    549 |         bool load_ok = elements.load(kplus, reorder->elements) && before.load(kplus, reorder->before);
07:46:48 
        |              ^~~~~~~
07:46:48 
  cc1plus: all warnings being treated as errors
```

Note the final three log-lines, in particular.

Comment (srl295) [reply core/src/ldml/ldml_transforms.cpp:549]: Unused because of the assert, right. Will fix.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.141-alpha](https://keyman.com/downloads/releases/alpha/17.0.141)

**Pull request merged**